webapp/channels/src/components/sidebar/sidebar.tsx

```diff
@@ -2,34 +2,36 @@
 // See LICENSE.txt for license information.
 
 import classNames from 'classnames';
-import React from 'react';
+import React, {lazy} from 'react';
 
 import {trackEvent} from 'actions/telemetry_actions';
 
-import BrowseChannels from 'components/browse_channels';
-import CreateUserGroupsModal from 'components/create_user_groups_modal';
+import {makeAsyncComponent} from 'components/async_load';
 import DataPrefetch from 'components/data_prefetch';
-import EditCategoryModal from 'components/edit_category_modal';
-import InvitationModal from 'components/invitation_modal';
-import KeyboardShortcutsModal from 'components/keyboard_shortcuts/keyboard_shortcuts_modal/keyboard_shortcuts_modal';
-import MoreDirectChannels from 'components/more_direct_channels';
-import NewChannelModal from 'components/new_channel_modal/new_channel_modal';
-import ResizableLhs from 'components/resizable_sidebar/resizable_lhs';
-import UserSettingsModal from 'components/user_settings/modal';
 
 import Pluggable from 'plugins/pluggable';
 import Constants, {ModalIdentifiers, RHSStates} from 'utils/constants';
-import * as Keyboard from 'utils/keyboard';
-import * as Utils from 'utils/utils';
+import {isKeyPressed, cmdOrCtrlPressed} from 'utils/keyboard';
+import {localizeMessage} from 'utils/utils';
 
 import type {ModalData} from 'types/actions';
 import type {RhsState} from 'types/store/rhs';
 
 import ChannelNavigator from './channel_navigator';
-import MobileSidebarHeader from './mobile_sidebar_header';
-import SidebarHeader from './sidebar_header';
 import SidebarList from './sidebar_list';
 
+const MobileSidebarHeader = makeAsyncComponent('MobileSidebarHeader', lazy(() => import('./mobile_sidebar_header')));
+const SidebarHeader = makeAsyncComponent('SidebarHeader', lazy(() => import('./sidebar_header')));
```

Comment:
I think it makes sense to lazily load the various modals, but I don't think we should be lazily loading the `SidebarHeader` or `ResizableLhs` since we want them to be immediately visible as the app loads

Reply:
reverted